autopush/websocket.py

```diff
@@ -270,21 +270,26 @@ def process_hello(self, data):
         self.uaid = uaid
 
         connect = data.get("connect")
-        if connect and self.ap_settings.pinger:
+        if connect and self.ap_settings.bridge:
             self.transport.pauseProducing()
-            d = deferToThread(self.ap_settings.pinger.register, uaid, connect)
+            d = deferToThread(self.ap_settings.bridge.register, uaid, connect)
             d.addCallback(self._check_router, True)
             d.addErrback(self.err_hello)
         else:
             self._check_router(False)
 
-    def _check_router(self, paused=False):
+    def _check_router(self, paused=False, bridge_register=None):
         if paused:
             self.transport.resumeProducing()
+        if bridge_register is not None and bridge_register is False:
```

Comment:
More of a philosophical question, but should we let the connection proceed even if bridge registration fails? This makes sense for some more than others: UDP wake-up is an optimization, and it's not critical if the `/netinfo/v1` endpoint is unavailable.

On the other hand, if bridge registration fails and the client can only use GCM/APNS, then it has a problem. But I think the client should decide that, and close the connection if it can't continue.